shop/internal/repository/repository_shop.go

```diff
@@ -0,0 +1,43 @@
+package repository
+
+import "github.com/GrabItYourself/giys-backend/lib/postgres/models"
+
+func (r *Repository) CreateShop(shop *models.Shop) error {
+	err := r.pg.Create(shop).Error
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Repository) GetShopById(id string) (*models.Shop, error) {
+	var shop models.Shop
+	err := r.pg.Where("id = ?", id).Take(&shop).Error
+	if err != nil {
+		return nil, err
+	}
+	return &shop, nil
+}
+
+func (r *Repository) EditShop(shop *models.Shop) (*models.Shop, error) {
+	err := r.pg.Model(&shop).Updates(shop).Error
+	if err != nil {
+		return nil, err
+	}
+	var edited_shop models.Shop
+	r.pg.Where("id = ?", shop.Id).Take(&edited_shop)
```

Comment:
To return updated values, use returning clauses.
https://gorm.io/docs/update.html#Returning-Data-From-Modified-Rows

but do we need the return value though? when we know the return value will be the same as `shop` input.

Reply:
Shop input can be only some fields (in the case that we want to update only partially) so I decided that we should return all of the shop.